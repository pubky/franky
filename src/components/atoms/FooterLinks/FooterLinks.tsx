import { cn } from '@/libs';
import { Typography } from '@/components';

export function FooterLinks({ children, className, ...props }: React.HTMLAttributes<HTMLParagraphElement>) {
  return (
    <Typography
      className={cn('text-muted-foreground opacity-80 font-normal text-sm leading-none', className)}
      {...props}
    >
      {children}
    </Typography>
  );
}
