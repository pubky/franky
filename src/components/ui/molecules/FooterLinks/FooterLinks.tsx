import { ReactNode } from 'react';
import { cn } from '@/libs';

interface LinkItem {
  text: string;
  href: string;
  target?: string;
}

interface FooterLinksProps {
  children: ReactNode;
  links?: LinkItem[];
  className?: string;
}

export function FooterLinks({ children, links = [], className }: FooterLinksProps) {
  return (
    <p className={cn('text-sm text-muted-foreground opacity-80', className)}>
      {children}
      {links.map((link, index) => (
        <a key={index} href={link.href} className="text-brand" target={link.target}>
          {link.text}
        </a>
      ))}
    </p>
  );
}
