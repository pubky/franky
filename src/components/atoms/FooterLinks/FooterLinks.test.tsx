import { describe, it, expect, vi } from 'vitest';
import { render, screen } from '@testing-library/react';
import { FooterLinks } from './FooterLinks';

// Mock UI components
vi.mock('@/components/ui', () => ({
  Typography: ({
    children,
    className,
    ...props
  }: {
    children: React.ReactNode;
    className?: string;
    [key: string]: unknown;
  }) => (
    <p data-testid="typography" className={className} {...props}>
      {children}
    </p>
  ),
}));

describe('FooterLinks', () => {
  it('renders with default props', () => {
    render(<FooterLinks>Footer text</FooterLinks>);

    const typography = screen.getByTestId('typography');
    expect(typography).toBeInTheDocument();
    expect(typography).toHaveClass('text-muted-foreground', 'opacity-80', 'font-medium', 'text-sm', 'leading-light');
    expect(screen.getByText('Footer text')).toBeInTheDocument();
  });

  it('applies custom className', () => {
    render(<FooterLinks className="custom-footer">Footer content</FooterLinks>);

    const typography = screen.getByTestId('typography');
    expect(typography).toHaveClass('custom-footer');
  });

  it('renders children correctly', () => {
    render(
      <FooterLinks>
        <span>Copyright 2024</span> | <a href="/privacy">Privacy</a>
      </FooterLinks>,
    );

    expect(screen.getByText('Copyright 2024')).toBeInTheDocument();
    expect(screen.getByText('Privacy')).toBeInTheDocument();
  });

  it('passes through additional props', () => {
    render(
      <FooterLinks data-testid="custom-footer-links" id="footer-id">
        Footer
      </FooterLinks>,
    );

    const typography = screen.getByTestId('custom-footer-links');
    expect(typography).toHaveAttribute('id', 'footer-id');
  });

  it('renders without children', () => {
    render(<FooterLinks />);

    const typography = screen.getByTestId('typography');
    expect(typography).toBeInTheDocument();
    expect(typography).toBeEmptyDOMElement();
  });
});

describe('FooterLinks - Snapshots', () => {
  it('matches snapshot with default props', () => {
    const { container } = render(<FooterLinks>Footer text</FooterLinks>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches snapshots for different configurations', () => {
    const { container: defaultContainer } = render(<FooterLinks>Default footer</FooterLinks>);
    expect(defaultContainer.firstChild).toMatchSnapshot();

    const { container: customContainer } = render(<FooterLinks className="custom-footer">Custom footer</FooterLinks>);
    expect(customContainer.firstChild).toMatchSnapshot();
  });

  it('matches snapshots for different children types', () => {
    const { container: textContainer } = render(<FooterLinks>Simple text</FooterLinks>);
    expect(textContainer.firstChild).toMatchSnapshot();

    const { container: complexContainer } = render(
      <FooterLinks>
        <span>Copyright 2024</span> | <a href="/privacy">Privacy</a>
      </FooterLinks>,
    );
    expect(complexContainer.firstChild).toMatchSnapshot();

    const { container: emptyContainer } = render(<FooterLinks />);
    expect(emptyContainer.firstChild).toMatchSnapshot();
  });

  it('matches snapshots for additional props', () => {
    const { container: withIdContainer } = render(<FooterLinks id="footer-id">Footer with ID</FooterLinks>);
    expect(withIdContainer.firstChild).toMatchSnapshot();

    const { container: withDataTestIdContainer } = render(
      <FooterLinks data-testid="custom-footer-links">Footer with test ID</FooterLinks>,
    );
    expect(withDataTestIdContainer.firstChild).toMatchSnapshot();
  });
});
