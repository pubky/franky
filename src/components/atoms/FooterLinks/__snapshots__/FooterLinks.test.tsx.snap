// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FooterLinks - Snapshots > matches snapshot with default props 1`] = `
<p
  class="text-muted-foreground opacity-80 font-medium text-sm leading-light"
  data-testid="typography"
>
  Footer text
</p>
`;

exports[`FooterLinks - Snapshots > matches snapshots for additional props 1`] = `
<p
  class="text-muted-foreground opacity-80 font-medium text-sm leading-light"
  data-testid="typography"
  id="footer-id"
>
  Footer with ID
</p>
`;

exports[`FooterLinks - Snapshots > matches snapshots for additional props 2`] = `
<p
  class="text-muted-foreground opacity-80 font-medium text-sm leading-light"
  data-testid="custom-footer-links"
>
  Footer with test ID
</p>
`;

exports[`FooterLinks - Snapshots > matches snapshots for different children types 1`] = `
<p
  class="text-muted-foreground opacity-80 font-medium text-sm leading-light"
  data-testid="typography"
>
  Simple text
</p>
`;

exports[`FooterLinks - Snapshots > matches snapshots for different children types 2`] = `
<p
  class="text-muted-foreground opacity-80 font-medium text-sm leading-light"
  data-testid="typography"
>
  <span>
    Copyright 2024
  </span>
   | 
  <a
    href="/privacy"
  >
    Privacy
  </a>
</p>
`;

exports[`FooterLinks - Snapshots > matches snapshots for different children types 3`] = `
<p
  class="text-muted-foreground opacity-80 font-medium text-sm leading-light"
  data-testid="typography"
/>
`;

exports[`FooterLinks - Snapshots > matches snapshots for different configurations 1`] = `
<p
  class="text-muted-foreground opacity-80 font-medium text-sm leading-light"
  data-testid="typography"
>
  Default footer
</p>
`;

exports[`FooterLinks - Snapshots > matches snapshots for different configurations 2`] = `
<p
  class="text-muted-foreground opacity-80 font-medium text-sm leading-light custom-footer"
  data-testid="typography"
>
  Custom footer
</p>
`;
